### ISSUE TITLE
Request: Torbox.app added as a service

### ISSUE THREAD

**Issue by Pukabyte** [debridmediamanager/zurg-testing#106]
**Feature Description**
- Adding Torbox.app as a service to zurg. Similar API to realdebrid and even has a translation guide as well.

**Problem and Motivation**
With real-debrid limiting connections and increasing rate limits, it is good to have other sources as backups. Torbox is a new debrid service that can cache Usenet files and torrents. It also seeds back on torrents which will help the whole ecosystem and keep debrid alive. I feel this is a necessary add and one that could really help out even if it is just a backup service for now until more people start to use it.

**Possible Solution**
The API documentation can be found here: https://api-docs.torbox.app and here: https://api.torbox.app/docs
The realdebrid translation guide can be found here:  https://www.postman.com/wamy-dev/workspace/torbox/collection/29572726-4244cdaf-ece6-4b6d-a2ca-463a3af48f54

**Preferred Timeline**
Whenever you can get to it if this is feasible and in your scope!

Remember to follow our contribution guidelines and code of conduct when submitting your request.

**Comment by TEALC82** [comment]
+1

**Comment by yowmamasita** [comment]
https://github.com/debridmediamanager/zurg-testing/issues/108#issuecomment-2201225456

Unfortunately have to close this because of this reason. zurg has already gone too far on the Real-Debrid rabbit hole.

**Comment by Pukabyte** [comment]
hmm with the new limitations, would you reconsider?

**Comment by frankgualdron** [comment]
+1

**Comment by yowmamasita** [comment]
I'm reopening to check interest level. I'm really loving Torbox developments recently

**Comment by Pukabyte** [comment]
Still very interested!

**Comment by b78691** [comment]
+1. Would love it if you could add support for Torbox.

**Comment by Cueball666uk** [comment]
+1 for Torbox support 👍🏻

**Comment by trulow** [comment]
[With today's news](https://x.com/RealDebrid/status/1859673163681960169?t=0SUJzMkUka5DGCdlnHaN7w), it may be time to rethink our options in regards to mounting Torbox

**Comment by JourneyOver** [comment]
Agreed, it would be nice to see torbox support due to today's news of whats going on with RD.

**Comment by thrashman69** [comment]
+1

**Comment by Arelav** [comment]
I see Torbox was added in DMM as a new login method.
I got a pretty good Black Friday deal for Premiumize. 
Is there are any plans to support Torbox in Zurg or both Torbox and Premiumize in DMM?
I can assist with sharing my account and/or testing.
I do not have access to closed beta, however if there are the plans to extend support I will consider joining.

**Comment by JourneyOver** [comment]
@Arelav last I heard he does have plans on adding it to zurg eventually, but he is currently getting things fully working with DMM first.

**Comment by Arelav** [comment]
@JourneyOver thanks, I hope it will work out. I'm ready to help if something can be useful from my side.

**Comment by petergriffiny** [comment]
+1 :)

**Comment by UnleashTheCode** [comment]
+1

**Comment by AdaptCharm** [comment]
+1

**Comment by Diego-DFC** [comment]
+1

**Comment by Pukabyte** [comment]
Welp, with the new direction Torbox has taken, dwindles the chance of this kind of stuff being supported.
Hopefully in future when things get developed more.

**Comment by Reasonable-Grape2698** [comment]
Torbox supports Webdav directly

**Comment by mot-stuff** [comment]
+1

**Comment by Reasonable-Grape2698** [comment]
docker-compose.yml
``` rclone:
    image: rclone/rclone:latest
    container_name: rclone
    restart: unless-stopped
    environment:
      TZ: Europe/Berlin
      PUID: 1000
      PGID: 1000
    volumes:
      - /mnt/torbox:/data:rshared # CHANGE /mnt/torbox WITH YOUR PREFERRED MOUNT PATH
      - ./rclone.conf:/config/rclone/rclone.conf
    cap_add:
      - SYS_ADMIN
    security_opt:
      - apparmor:unconfined
    devices:
      - /dev/fuse:/dev/fuse:rwm
    command: "mount torbox: /data --allow-other --allow-non-empty --dir-cache-time 10s --vfs-cache-mode full"
```

**rclone.conf:**
```[torbox]
type = webdav
url = https://webdav.torbox.app/
vendor = other
user = email@example.com
pass = {encryptedPassword}
```

rclone uses base64 encryption, you can obscure a password using `rclone obscure "(password)"`, in docker:
```
docker run -it rclone/rclone:latest obscure "(password)"
```


this is a flat directory and doesn't have movies/tvshows folders like zurg, but works without any middlewear.

**Comment by Reasonable-Grape2698** [comment]
It appears Torbox has an availability endpoint that accepts unlimited hashes at once (comma separated), meaning you can curl and check up to 150 at a time. This could be useful for DMM

https://github.com/Reasonable-Grape2698/torbox_dmm_availability_scanner

Tacky code that checks a dmm.json backup and outputs to output.txt if the hash is cached on Torbox, then add.sh can be used to add them. As you can do 150 at a time there's no rate limiting for checking torrents.

**Comment by kvnkim** [comment]
Torbox's current implementation of WebDAV is very slow and unresponsive and times out when using rclone. Using FTP/WebDAV methods both have connectivity issues. Zurg's WebDAV for R-D is working flawlessly as of now. Hoping for zurg to add its own implementation of a fast working WebDAV someday.

**Comment by Cueball666uk** [comment]
Torbox has recently updated WebDAV and it's now crazy fast from my testing.

**Comment by berot3** [comment]
Seems zurg is at a dead end? What are the alternatives?

**Comment by Arelav** [comment]
> Seems zurg is at a dead end? What are the alternatives?

I don't agree with the statement. It's more like it's RD only and perhaps will stay this way. 
You can take a look on Riven, Torbox-Media-Centere (look on TB repo on GH, it's official one) and Decypharr. This what I'm aware of. If you Jellyfin user it's Gelato for Jellyfin (different but may be useful for you - https://github.com/awesome-jellyfin/awesome-jellyfin)

**Comment by berot3** [comment]
Thx. Yes, TBMC is the way to got, from what I can see.

**Comment by Arelav** [comment]
It's was working for me but sometimes it was choking whole system that my Synology NAS was barely responding on ssh if I was patient enough to wait an kill the process and even this wasn't easy task. One time I had to reset the system using a physical button. Otherwise, especially in Jellyfin with strm files used instead of fuse it was great.
